Rollbar/RollbarNotifier.m

```diff
@@ -462,6 +472,109 @@ - (NSString*)generateUUID {
     return string;
 }
 
+#pragma mark - Payload truncate
+
+- (void)createMutablePayloadWithData:(NSMutableDictionary *)data forPath:(NSString *)path {
+    NSArray *pathComponents = [path componentsSeparatedByString:@"."];
+    NSString *currentPath = @"";
+
+    for (int i=0; i<pathComponents.count; i++) {
+        NSString *part = pathComponents[i];
+        currentPath = i == 0 ? part : [NSString stringWithFormat:@"%@.%@", currentPath, part];
+        id val = [data valueForKeyPath:currentPath];
+        if (!val) return;
+        if ([val isKindOfClass:[NSArray class]] && ![val isKindOfClass:[NSMutableArray class]]) {
+            NSMutableArray *newVal = [NSMutableArray arrayWithArray:val];
+            [data setValue:newVal forKeyPath:currentPath];
+        } else if ([val isKindOfClass:[NSDictionary class]] && ![val isKindOfClass:[NSMutableDictionary class]]) {
+            NSMutableDictionary *newVal = [NSMutableDictionary dictionaryWithDictionary:val];
+            [data setValue:newVal forKeyPath:currentPath];
+        }
+    }
+}
+
+- (void)truncatePayload:(NSMutableDictionary *)data forKeyPath:(NSString *)keypath {
+    NSData *jsonPayload = [NSJSONSerialization dataWithJSONObject:data options:0 error:nil safe:true];
+    NSInteger dataSize = jsonPayload.length * 0.001;
+
+    if (dataSize <= MAX_PAYLOAD_SIZE) {
+        return;
+    }
+
+    [self createMutablePayloadWithData:data forPath:keypath];
+    NSMutableArray *array = [data valueForKeyPath:keypath];
+    [data setValue:@[] forKeyPath:keypath];
+
+    jsonPayload = [NSJSONSerialization dataWithJSONObject:data options:0 error:nil safe:true];
+    NSInteger sizeDiff = dataSize - jsonPayload.length * 0.001;
+
+    double sizePerItem = sizeDiff / (double)array.count;
+    if (dataSize > MAX_PAYLOAD_SIZE) {
```

Comment:
What is the point of this `if` it is necessarily true if we get here because of 500-502

Reply:
Ah, yes. This was left over from testing. Removed